### ISSUE TITLE
DicomDataset.DoAdd() unable to get corresponding VR based on value type

### ISSUE THREAD

**Issue by apang2013** [fo-dicom/fo-dicom#1068]
**TL;DR:
I believe `DoAdd<T>(DicomTag tag, Encoding encoding, IList<T> values, bool allowUpdate)` is not properly checking the other types of the VR that matches the type of the input. For DicomTag.VR.Count > 1, it is currently safest to explicitly state the desired VR, but the user shouldn't be compelled to do so just to avoid this bug.**

### Expected behavior
Most Dicom files have PixelData with 
```
BitsAllocated = 16
BitsStored = 12
HighBit = 11
```
Which specifies 12 bit data, since `byte` is insufficient, we use `ushort`; consider the following (modified from examples/PrintSCU)
```
DicomDataset dataset = new DicomDataset();

dataset.Add<ushort>(DicomTag.Columns, (ushort)16)
                .Add<ushort>(DicomTag.Rows, (ushort)256)
                .Add<ushort>(DicomTag.BitsAllocated, 16)
                .Add<ushort>(DicomTag.BitsStored, 12)
                .Add<ushort>(DicomTag.HighBit, 11)
                .Add(DicomTag.PixelRepresentation, (ushort)PixelRepresentation.Unsigned)
                .Add(DicomTag.PlanarConfiguration, (ushort)PlanarConfiguration.Interleaved)
                .Add<ushort>(DicomTag.SamplesPerPixel, 1)
                .Add(DicomTag.PhotometricInterpretation, PhotometricInterpretation.Monochrome2.Value);
            
ushort[] data = GetTestData();

dataset.Add<ushort>(DicomVR.OW, DicomTag.PixelData, null, data); // Does not throw error
dataset.Add<ushort>(DicomTag.PixelData, data); // Throws error
...

ushort[] GetTestData()
        {
            int dimensions = 4096 * 1;
            ushort[] bits = new ushort[dimensions];
            for (int i = 0; i < dimensions; i++)
            {
                bits[i] = (ushort)(i);
            }
            return bits;
        }
```
Both overloads of Add() should be OK, since PixelData's VR is OB/OW, OB indeed only allows 8 bit data, but OW is supposed to allow ushort's (16).

### Actual behavior

When using the `dataset.Add<ushort>(DicomTag.PixelData, data);` FoDicom would throw the following:
```
An unhandled exception of type 'System.InvalidOperationException' occurred in Dicom.Core.dll
Unable to create DICOM element of type OB with values of type System.UInt16
```

I believe the bug is given in `DoAdd<T>(DicomTag tag, Encoding encoding, IList<T> values, bool allowUpdate)`:
```
if (values != null) vr = entry.ValueRepresentations.FirstOrDefault(x => x.ValueType == typeof(T));
            if (vr == null) vr = entry.ValueRepresentations.First();

            return DoAdd(vr, tag, encoding, values, allowUpdate);
```
Which is mistakenly taking the first VR (which is OB); yet my input is `ushort`, which means it should be passing the VR as OW (the second VR) instead.
Subsequently, causing : `DicomDataset.DoAdd<T>(DicomVR vr, DicomTag tag, Encoding encoding, IList<T> values, bool allowUpdate) ` to throw the error, because it evaluates my `ushort[]` as if a `byte[]`:
```
            if (vr == DicomVR.OB)
            {
                if (values == null) return DoAdd(new DicomOtherByte(tag, EmptyBuffer.Value), allowUpdate);
                if (typeof(T) == typeof(byte)) return DoAdd(new DicomOtherByte(tag, values.Cast<byte>().ToArray()), allowUpdate);

                if (typeof(T) == typeof(IByteBuffer) && values.Count == 1)
                {
                    return DoAdd(new DicomOtherByte(tag, (IByteBuffer)values[0]), allowUpdate);
                }
            }
```

### Steps to reproduce the behavior
Attempt create a dicom with 12 bit pixeldata (see the sample above).

I suspect any DoAdd function where a DicomTag.VR's > 1 would be susceptible to the same bug, simply because the data type is interpreted incorrectly, and would default to the first VR, at least for this example it seems to be the case.

### fo-dicom version and OS/platform
4.0.5, Windows 10, c#

**Comment by gofal** [comment]
This is a special case that appeals only to PixelData and not to all Tags with VR>1, because in other cases the VR is unique for a DicomTag. Only PixelData has the two possibilities of OB or OW.
The `DoAdd` method is currently isolated, that means it does not depend on the dataset that is already present. Fo-dicom tried to come around this problem by introducing the class `DicomPixelData` that wraps a `DicomDataset` and has the method `AddFrame`, where you can pass in any ByteBuffer and the DicomPixelData class then interprets it depending on values of `bitsAllocated`.

**Comment by gofal** [comment]
To make it more obvious to the user, maybe the DicomDataset should get a method `AddPixelData` which then takes various arrays and checks the dataset how to convert/interpret this array?
